homeassistant/components/ihc/light.py

```diff
@@ -6,47 +6,43 @@
 from ihcsdk.ihccontroller import IHCController
 
 from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
 from .const import CONF_DIMMABLE, CONF_OFF_ID, CONF_ON_ID, DOMAIN, IHC_CONTROLLER
 from .ihcdevice import IHCDevice
 from .util import async_pulse, async_set_bool, async_set_int
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
-    """Set up the IHC lights platform."""
-    if discovery_info is None:
-        return
-    devices = []
-    for name, device in discovery_info.items():
-        ihc_id = device["ihc_id"]
-        product_cfg = device["product_cfg"]
-        product = device["product"]
-        # Find controller that corresponds with device id
-        controller_id = device["ctrl_id"]
-        ihc_controller: IHCController = hass.data[DOMAIN][controller_id][IHC_CONTROLLER]
-        ihc_off_id = product_cfg.get(CONF_OFF_ID)
-        ihc_on_id = product_cfg.get(CONF_ON_ID)
-        dimmable = product_cfg[CONF_DIMMABLE]
-        light = IhcLight(
-            ihc_controller,
-            controller_id,
-            name,
-            ihc_id,
-            ihc_off_id,
-            ihc_on_id,
-            dimmable,
-            product,
-        )
-        devices.append(light)
-    add_entities(devices)
+    """Load IHC lights based on a config entry."""
+    controller_id: str = str(entry.unique_id)
+    controller_data = hass.data[DOMAIN][entry.entry_id]
+    ihc_controller: IHCController = controller_data[IHC_CONTROLLER]
+    lights = []
+    if "light" in controller_data and controller_data["light"]:
+        for name, device in controller_data["light"].items():
```

Comment:
Can be simplified to:

```suggestion
    for name, device in controller_data.get("light",{}).items():
```